sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala

```diff
@@ -132,4 +134,8 @@ private[sql] object CatalogV2Implicits {
       part
     }
   }
+
+  def unquote(name: String): Seq[String] = {
+    new CatalystSqlParser(SQLConf.get).parseMultipartIdentifier(name)
```

Comment:
This shouldn't create a new parser each time it is used. Can you create one in a companion object?

Reply:
Also, is there a better place for this? If there is a common need for a parser for names, then maybe the `SparkSqlParser` object should expose methods to parse names and have an internal instance?

Reply:
How about 

```scala
  private lazy val catalystSqlParser = new CatalystSqlParser(SQLConf.get)

  def unquote(name: String): Seq[String] = {
    catalystSqlParser.parseMultipartIdentifier(name)
  }
```
for now until we find a good place to have an internal instance of `catalystSqlParser`?

The concern is that if we have it in the singleton, what if the `SQLConf` is changed?